azuredevops/internal/service/serviceendpoint/resource_serviceendpoint_artifactory.go

```diff
@@ -144,9 +144,13 @@ func flattenServiceEndpointArtifactory(d *schema.ResourceData, serviceEndpoint *
 			if len(authMap) > 0 {
 				newHash, hashKey := tfhelper.HelpFlattenSecretNested(d, "authentication_token", authMap, "token")
 				auth[hashKey] = newHash
-				auth["token"] = authMap["token"]
 			}
 		}
+
+		if serviceEndpoint.Authorization != nil && serviceEndpoint.Authorization.Parameters != nil {
```

Comment:
Service will always return the `token` in empty string or null. The `token` should get from the `state` or HCL instead of service response.

Reply:
I can see that in this PR(https://github.com/microsoft/terraform-provider-azuredevops/pull/705), the author has made the same changes as me. Does that mean that he is wrong as well ? If so, I will revert my changes and wait for a better fix to come along.

Reply:
In this resource the `token`  was encoded and set it to empty string, when users update any other properties of this resource and apply it, the new `token` which is  empty string  will be send to the service 

Also it is recommend to remove the hash of the `token/password` and these functions has be deprecated.

Reply:
I've reverted my changes and will wait for the right fix.